### DISCUSSION TITLE
should it work when the project is not a lib

### DISCUSSION THREAD

**Discussion by jiangxiaoqiang** [diesel-rs/diesel#2878] [Q&A]
when I am using this code to import the package in my project in the `main.rs`:

```
#[macro_use] 
extern crate reddwarf_music;

fn main(){
    
}
```

shows that:

```
   Compiling reddwarf_music v0.1.0 (/Users/dolphin/Documents/GitHub/reddwarf_music)
error[E0463]: can't find crate for `reddwarf_music`
 --> src/main.rs:6:1
  |
6 | extern crate reddwarf_music;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `reddwarf_music`

To learn more, run the command again with --verbose.
(base) 
```

must I using a lib? not work in binary package project? This is my Cargo.toml:

```

[package]
name = "reddwarf_music"
version = "0.1.0"
edition = "2018"

[dependencies]
rocket = { version = "0.5.0-rc.1", features = ["json"] }
rand = "0.8.4"
serde = { version = "1.0.64", features = ["derive"] }
serde_json = "1.0.64"
reqwest = "0.11.4"

# database
diesel = { version = "1.4.4", features = ["postgres"] }
dotenv = "0.15.0"

```

**Comment by weiznich** [comment]
Please stop opening multiple discussion threads at once, especially if they are not related to diesel at all (like this question). I suggest that you start with the rust book to learn a bit about the general rust project setup. After that the diesel getting started guide contains a example project and all information required to fix your issues (as well as other important information)

**Reply by jiangxiaoqiang** [reply]
sorry, I am learning rust, and follow the docs to make some CRUD to Postgres Database. But the question could not be able to delete. I am sorry to ask questiones out of diesel.
